qoffer-infrastructure/src/main/groovy/life/qbic/portal/qoffer2/web/views/SearchCustomerView.groovy

```diff
@@ -0,0 +1,256 @@
+        super()
+        this.viewModel = viewModel
+        this.searchCustomerViewModel = searchCustomerViewModel
+        this.foundCustomerList = searchCustomerViewModel.customers
+        initLayout()
+        setupDataProvider()
+        setupFieldValidators()
+        registerListeners()
+    }
+
+    /**
+     * Generates a vaadin Form Layout as an UserInterface consisting of vaadin components
+     * to enable user input for Customer creation
+     */
+    private def initLayout() {
+
+
+        //Generate FormLayout and the individual components
+        FormLayout searchCustomerForm = new FormLayout()
+
+        this.firstNameField = new TextField("First Name")
+        firstNameField.setPlaceholder("First Name")
+        firstNameField.setRequiredIndicatorVisible(true)
+
+        this.lastNameField = new TextField("Last Name")
+        lastNameField.setPlaceholder("Last Name")
+        lastNameField.setRequiredIndicatorVisible(true)
+
+        this.submitButton = new Button("Search Customer")
+        submitButton.setIcon(VaadinIcons.SEARCH)
+
+        this.clearButton = new Button("Clear Customers")
+        clearButton.setIcon(VaadinIcons.CLOSE_CIRCLE)
+        clearButton.setEnabled(false)
+
+        this.customerGrid = new Grid<Customer>()
+        this.gridLayout = new HorizontalLayout()
+
+        HorizontalLayout row1 = new HorizontalLayout(firstNameField, lastNameField)
+        row1.setSizeFull()
+        row1.setDefaultComponentAlignment(Alignment.BOTTOM_LEFT)
+
+        HorizontalLayout row2 = new HorizontalLayout(clearButton, submitButton)
+        row2.setSizeFull()
+        row2.setDefaultComponentAlignment(Alignment.BOTTOM_LEFT)
+
+        HorizontalLayout row3 = new HorizontalLayout(gridLayout)
+        row3.setSizeFull()
+        row3.setDefaultComponentAlignment(Alignment.BOTTOM_LEFT)
+
+        //Add the components to the FormLayout
+        searchCustomerForm.addComponent(row1)
+        searchCustomerForm.addComponent(row2)
+        searchCustomerForm.addComponent(row3)
+
+        firstNameField.setSizeFull()
+        lastNameField.setSizeFull()
+
+        this.setSpacing(true)
+        this.addComponent(searchCustomerForm)
+
+    }
+
+    /**
+     * This method calls the individual Listener generation methods
+     */
+
+    private def registerListeners() {
+
+        submitButton.addClickListener({ event ->
+            submitCustomer()
+
+        })
+        clearButton.addClickListener({ event ->
+            clearSearch()
+            //clear the search fields
+            firstNameField.clear()
+            lastNameField.clear()
+            firstNameField.setComponentError(null)
+            lastNameField.setComponentError(null)
+        })
+    }
+
+    /**
+     * This method adds the retrieved Customer Information to the Customer grid
+     */
+    private void setupDataProvider() {
+
+        this.customerGrid.setItems(foundCustomerList)
+    }
```

Comment:
NIT: why creating a function for just one line of code?

Reply:
The general View structure was adapted from the CreateCustomerView and while this function may be overkill in this view, i think it's good practice to adhere to a coherent setup flow(initlayout add data etc.) for all views for easier understanding.